cmd/metadata-server/server.go

```diff
@@ -154,34 +181,76 @@ func (s *Server) GetDirectoryKeys(ctx context.Context, req *mpb.GetDirectoryKeys
 		return nil, err
 	}
 
-	keys := make([]string, 0, len(metadata.Entries))
-	for k := range metadata.Entries {
-		keys = append(keys, k)
+	entries := []*mpb.GetDirectoryEntriesResponse_DirectoryEntry{}
+	for fpath, entry := range metadata.Entries {
```

Comment:
very minor, but fpath can refer to directory paths as well so might as well stick with the blob term for consistency